jest.config.ts

```diff
@@ -0,0 +1,88 @@
+import type {Config} from "jest";
+import os from "node:os";
+import pathPosix from "node:path/posix";
+
+
+let PRIMARY_REPORTER: string | [string, Record<string, unknown>] = "default";
+if ("undefined" !== typeof process.env.CI) {
+    PRIMARY_REPORTER = [
+        "github-actions",
+        {silent: false},
+    ];
+}
+console.log(`Environment variable "CI"="${process.env.CI}": ` +
+    `primary reporter will be ${JSON.stringify(PRIMARY_REPORTER)}`);
+
+const JEST_CONFIG: Config = {
+    collectCoverage: true,
+    collectCoverageFrom: ["src/**/*.{ts,tsx}"],
+    coverageProvider: "v8",
+    coverageReporters: ["text"],
+    coverageThreshold: {
+        "global": {
+            functions: 90,
+            lines: 90,
+        },
+        // eslint-disable-next-line no-warning-comments
+        // TODO: Remove / Adjust below overrides as more test cases are added.
+        "src/*.tsx": {
+            functions: 0,
+            lines: 0,
+        },
+        "src/components/": {
+            functions: 0,
+            lines: 0,
+        },
+        "src/contexts/": {
+            functions: 0,
+            lines: 0,
+        },
+        "src/services/": {
+            functions: 0,
+            lines: 0,
+        },
+        "src/typings/": {
+            functions: 0,
+            lines: 0,
+        },
+        "src/utils/": {
+            functions: 0,
+            lines: 0,
+        },
+        "src/utils/math.ts": {
+            functions: 100,
+            lines: 100,
+        },
+    },
+    displayName: {
+        name: "yscope-log-viewer",
+        color: "blue",
+    },
+    errorOnDeprecated: true,
+    maxConcurrency: os.cpus().length,
+    maxWorkers: "100%",
+    openHandlesTimeout: 1000,
+    reporters: [
+        PRIMARY_REPORTER,
+        "summary",
+    ],
+    showSeed: true,
+    testEnvironment: "node",
```

Comment:
This is the default already, right?

Reply:
Right, I was thinking we might need to switch it to `jsdom` in the future, once we enable snapshot testing for the UI code. Similarly, we might want to switch the `coverageProvider` back to `bazel` if we see issues with the UI code.
For now, i can remove as well. What do you think?

Reply:
* `bazel` 😛 
* Yeah, I think we can remove it for now.

Reply:
that was a typo 🤣. I meant "babel"